pkg/services/ngalert/state/state.go

```diff
@@ -113,16 +114,12 @@ func (a *State) resultNoData(alertRule *ngModels.AlertRule, result eval.Result)
 }
 
 func (a *State) NeedsSending(resendDelay time.Duration) bool {
-	if a.State != eval.Alerting && a.State != eval.Normal {
-		return false
-	}
-
-	if a.State == eval.Normal && !a.Resolved {
+	if a.State == eval.Pending || a.State == eval.Error || a.State == eval.Normal && !a.Resolved {
```

Comment:
I think I understand the intention of this change but I just wanted to check that what we are doing is changing `a.State != eval.Alerting && a.State != eval.Normal` to `a.State == eval.Pending || a.State == eval.Error` such that `eval.NoData` will return true (depending on `LastEvaluationTime`).

Reply:
Correct, the previous logic always returned false to NoData. I merged two conditions and rewrote with equivalent disjunction which is a little bit more readable as it does not use negations. Then I just removed NoData from the expression.